frontend/packages/dev-console/src/components/edit-application/EditApplication.tsx

```diff
@@ -31,8 +36,16 @@ const EditApplication: React.FC<EditApplicationProps> = ({
   const { t } = useTranslation();
   const [perspective] = useActivePerspective();
   const perspectiveExtensions = useExtensions<Perspective>(isPerspective);
+  const uploadJarFormToastCallback = useUploadJarFormToast();
   const initialValues = getInitialValues(appResources, appName, namespace);
-  const pageHeading = getPageHeading(_.get(initialValues, 'build.strategy', ''));
+  const pageHeading = getPageHeading(
+    _.get(initialValues, 'build.strategy', ''),
+    _.get(initialValues, 'build.source.type', undefined),
+  );
+  const validationSchema = getValidationSchema(
+    _.get(initialValues, 'build.strategy', ''),
+    _.get(initialValues, 'build.source.type', undefined),
+  );
```

Comment:
nit: does it make sense

```suggestion
 const buildStrategy = initialValues.build?.strategy ??  ' ';
 const buildSourceType = initialValues.build?.source?.type;
  const pageHeading = getPageHeading(buildStrategy, buildSourceType);
  const validationSchema = getValidationSchema(buildStrategy, buildSourceType);
```